### PR TITLE
Support for T.inverse() and R.inverse()

### PR DESCRIPTION
**Pull request by cfo**
As discussed with @simonlynen, it would be nice to support inverse() as this is also the wording used in Eigen.

### CODE REVIEW

**Comment by mikebosse** [comment]
I don't see the point in duplicating the `inverted` functions. The code is redundant and will lead to more confusions.

**Comment by cfo** [comment]
@mikebosse  I agree it will lead to confusions to have both. I just found the name inverted() a bit strange but if you think we should stick to it, that's no problem.

**Comment by mikebosse** [comment]
I'm not particular on which version is used, `inverse` may read better, but then there is a lot of code already written that will need to be updated.
@HannesSommer @gehrinch what is your input on the naming convention?

**Comment by cfo** [comment]
Maybe I was a bit over-hasty. To the authors of Kindr: just close the pull request if you think that's a useless discussion.

**Comment by simonlynen** [comment]
We can consider switching our code to this as it would avoid errors due to people being used to what Eigen does.

**Comment by HannesSommer** [comment]
I would support this switch as this deviation from Eigen's usage was a unfortunate decision in the first place - in my opinion. I cannot recall a strong reason. At max the similarity to ```invert()```, which is the inplace inversion.  @gehrinch or @remod, do you remember a  good or even _the_ reason?

For minkindr we could have transition phase having both and using a deprecation declaration (see e.g. http://stackoverflow.com/a/21265197). Or do we have already something better? I guess till we switch to c++14 we can live with such a macro.

For kindr we could switch with the upcoming major interface revision. In fact I was thinking of that already there.

**Comment by simonlynen** [comment]
@mikebosse how you use minkindr? Is this a submodule in your repo or do you clone it?

**Comment by mikebosse** [comment]
it is cloned for use with the continuous scan matching and curves

**Comment by gehrinch** [comment]
For me, it was not fully clear if inverse() does change the data or just returns an inverted copy. If I'm right, inverse can also be used as a verb, which would mean that the data is modified.  Therefore invered() and invert() is clearer in my opinion. What does native English speaker think?

**Comment by gehrinch** [comment]
Better naming would be: getInverse()

**Comment by HannesSommer** [comment]
Hm, this sounds like doing it _ideally_ vs. doing it _as usual_ (here pretty much given by Eigen, which is very well known by potential (min)kindr users).

What about not having the inplace operations in the new concept of kindr and then migrate to the Eigen like ```.inverse()```? 

How important are the inplace operations? In my (recent) opinion they are not worth the danger (side effects, confusion) and burden (extra bloat, naming difficulties, see this discussion ;) ) they bring. Performance wise they probably deliver almost nothing in average (after low level optimization) - especially if we keep the return type open to different representations, which really can boost performance in some occasions.

**Comment by HannesSommer** [comment]
@cfo , about the actual changes: you copied non structural code (the body). That is a very bad idea :). If we have both for some transition phase then it should be one (ideally the deprecated one if there is one) calling the other (or both calling a third, common one)!

**Comment by simonlynen** [comment]
@cfo please change the CL to the following:
Rename the inverted to inverse. Add an inverted method and deprecate it, let it call the inverse method.

**Comment by simonlynen** [comment]
ok to test

**Comment by cfo** [comment]
@simonlynen ok?

File: minkindr/include/kindr/minimal/quat-transformation.h [hunk 1]
```diff
@@ -58,6 +58,7 @@ class QuatTransformationTemplate {
   typedef Eigen::Matrix<Scalar, 3, 3> RotationMatrix;
   typedef Eigen::Matrix<Scalar, 4, 4> TransformationMatrix;
 
+  /// \brief Constructor of identity tranformation.
```
Comment (simonlynen) [inline minkindr/include/kindr/minimal/quat-transformation.h]: transformation

File: minkindr/test/test-transformation.cc [hunk 2]
```diff
@@ -151,17 +151,17 @@ TEST(MinKindrTests, testCompose) {
...
 
   Transformation T(q1,t);
-  Transformation invT = T.inverted();
+  Transformation invT = T.inverse();
   Eigen::Vector3d v(6.26257419,  1.58356548,  6.05772983);
   Eigen::Vector3d invTTv(-8.16137069,  -6.14469052, -14.34176544);
   Eigen::Vector3d TTv(5.52009598,  24.34170933,  18.9197339);
 
   
   Transformation TT = T*T;
```
Comment (simonlynen) [inline minkindr/test/test-transformation.cc]: spaces around *

File: minkindr/test/test-transformation.cc [hunk 3]
```diff
@@ -151,17 +151,17 @@ TEST(MinKindrTests, testCompose) {
...
-  Transformation Id1 = T*T.inverted();
-  Transformation Id2 = T.inverted() * T;
-  Transformation iTiT1 = T.inverted() * T.inverted();
-  Transformation iTiT2 = TT.inverted();
+  Transformation Id1 = T*T.inverse();
```
Comment (simonlynen) [inline minkindr/test/test-transformation.cc]: sp around *

File: minkindr_gtsam/test/test_minkindr-gtsam.cc [hunk 4]
```diff
@@ -647,7 +647,7 @@ TEST(MinkindrGtsamTests, testCubicHermiteQuaternionDerivative) {
     Eigen::Vector3d vI = interpV.value(values);
 
     EXPECT_TRUE(EIGEN_MATRIX_NEAR(vI,waVal,1e-3));
-    Eigen::Vector3d dq = (qI * qaVal.inverted()).log()/1e-5;
+    Eigen::Vector3d dq = (qI * qaVal.inverse()).log()/1e-5;
```
Comment (simonlynen) [inline minkindr_gtsam/test/test_minkindr-gtsam.cc]: sp around /

File: minkindr_gtsam/test/test_minkindr-gtsam.cc [hunk 5]
```diff
@@ -660,7 +660,7 @@ TEST(MinkindrGtsamTests, testCubicHermiteQuaternionDerivative) {
       EQuaternion interpQp = hermiteInterpolation(qA, wA, qB, wB, alpha+fd_step);
       EVector3 interpV = hermiteInterpolationDerivative(qA, wA, qB, wB, alpha);
 
-      Eigen::Vector3d dq = (interpQp.value(values) * interpQ.value(values).inverted()).log()/fd_step;
+      Eigen::Vector3d dq = (interpQp.value(values) * interpQ.value(values).inverse()).log()/fd_step;
```
Comment (simonlynen) [inline minkindr_gtsam/test/test_minkindr-gtsam.cc]: sp around /

**Comment by simonlynen** [comment]
lgtm. Thanks.

**Pull request merged**